### PR TITLE
Connector protocol

### PR DESCRIPTION
**Pull request by boaks**
Introduce protocol in Connector.
Remove isSecure from RawData.

See issue #156 

Signed-off-by: Achim Kraus <achim.kraus@bosch-si.com>

### CODE REVIEW

**Comment by boaks** [comment]
Alternatively we may use an enumeration in Connector instead of String as protocol.

File: element-connector/src/test/java/org/eclipse/californium/elements/UDPConnectorTest.java [hunk 1]
```diff
@@ -49,10 +49,8 @@ public void stop() {
 	}
 
 	@Test
-	public void testGetUriContainsCorrectSchemeAndAddress() {
-		assertThat(connector.getUri().getScheme(), is("coap"));
-		assertThat(connector.getUri().getHost(), is(connector.getAddress().getHostString()));
-		assertThat(connector.getUri().getPort(), is(connector.getAddress().getPort()));
+	public void testGetProtocol() {
+		assertThat(connector.getProtocol(), is("UDP"));
```
Comment (sbernard31) [inline element-connector/src/test/java/org/eclipse/californium/elements/UDPConnectorTest.java]: Not sure this test make so much sense

File: scandium-core/src/test/java/org/eclipse/californium/scandium/DTLSConnectorTest.java [hunk 2]
```diff
@@ -245,10 +245,8 @@ private static DtlsConnectorConfig newStandardConfig(InetSocketAddress bindAddre
 	 * Verifies that the connector's endpoint URI corresponds to the configuration.
 	 */
 	@Test
-	public void testGetUriContainsCorrectSchemeAndAddress() {
-		assertThat(server.getUri().getScheme(), is("coaps"));
-		assertThat(server.getUri().getHost(), is(serverEndpoint.getHostString()));
-		assertThat(server.getUri().getPort(), is(serverEndpoint.getPort()));
+	public void testGetProtocol() {
+		assertThat(server.getProtocol(), is("DTLS"));
```
Comment (sbernard31) [inline scandium-core/src/test/java/org/eclipse/californium/scandium/DTLSConnectorTest.java]: same remarks. (if we keep it, comment is not ok)

**Comment by sbernard31** [comment]
> Alternatively we may use an enumeration in Connector instead of String as protocol.

enum seems overkill for me.

**Pull request merged**